sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -1222,7 +1239,244 @@ void set(
         throws SQLException;
   }
 
-  /** A {@link PTransform} to write to a JDBC datasource. */
+  /**
+   * A {@link PTransform} to write to a JDBC datasource. Executes statements one by one.
+   *
+   * <p>The INSERT, UPDATE, and DELETE commands sometimes have an optional RETURNING clause that
+   * supports obtaining data from modified rows while they are being manipulated. Output {@link
+   * PCollection} of this transform is a collection of such returning results mapped by {@link
+   * RowMapper}.
+   */
+  @AutoValue
+  public abstract static class WriteWithResults<T, V>
+      extends PTransform<PCollection<T>, PCollection<V>> {
```

Comment:
Would it better to wrap up `V` into Beam abstract class or interface, e.g. `JdbcWriteResult`?

Reply:
Add abstract class `JdbcWriteResult` and now `WriteWithResult` class uses `<T, V extends JdbcWriteResult>`.